First pass at List View

jfsiii opened the pull request:
## Summary

<img width="1560" alt="Screen Shot 2019-07-28 at 3 42 15 PM" src="https://user-images.githubusercontent.com/57655/62011993-4e41a080-b14e-11e9-81cb-3e93acb96917.png">

![IM list to detail](https://user-images.githubusercontent.com/57655/62012253-f86ef780-b151-11e9-9e18-ef4c49564efa.gif)

_some other notes inline but highlights are_

### Visual changes
 * Uses 4 column layout from Figma designs where [left column is for the Integration status](https://github.com/elastic/kibana/blob/db244c9d8fa4350286fdbc686b97949e5fb06298/x-pack/legacy/plugins/integrations_manager/public/components/integration_list_grid.tsx#L51-L55) and (eventually) category and [the right 3 are for the integration cards](https://github.com/elastic/kibana/blob/db244c9d8fa4350286fdbc686b97949e5fb06298/x-pack/legacy/plugins/integrations_manager/public/components/integration_list_grid.tsx#L56-L63)
 * Header background is grey with 1px bottom border
 * Header copy is [centered vertically](https://github.com/elastic/kibana/blob/5d944f77c26539325141e62ccbafb1588b5d6ce6/x-pack/legacy/plugins/integrations_manager/public/screens/home.tsx#L67). No search box at the moment because that feature isn't yet available (and might not be in our initial release)
 * Header image is [anchored to the bottom](https://github.com/elastic/kibana/blob/5d944f77c26539325141e62ccbafb1588b5d6ce6/x-pack/legacy/plugins/integrations_manager/public/screens/home.tsx#L86).

### Architecture changes

  * Created [`PluginCore`](https://github.com/elastic/kibana/blob/5d944f77c26539325141e62ccbafb1588b5d6ce6/x-pack/legacy/plugins/integrations_manager/public/plugin.tsx#L18-L22) (replacing `CoreStart`)
  * Load routes in `public/index` and pass them to the plugin [via the `.start` lifecycle](https://github.com/elastic/kibana/blob/5d944f77c26539325141e62ccbafb1588b5d6ce6/x-pack/legacy/plugins/integrations_manager/public/index.ts#L30-L33) vs loading them in `public/plugin`
  * Use New Platform's `chrome.setBreadcrumbs` method to set breadcrumbs
  * Added a `useCore` hook to make it easier to consume items from the core context
  * Added a `useBreadcrumbs` hook (which calls `useCore`) so views don't need to know how breadcrumbs are set. Room to improve / iterate here (add dependency keys to, allow appending of keys vs setting, etc)
  * Created a `PLUGIN` constant and [replaced `PLUGIN_ID` with `PLUGIN.ID`](7423a3897ea7acb9c68afb125ce67fd22b2e522b) (as many other plugins do). Moved some other [plugin-related values under `PLUGIN`](41fb11be8e799f5140dc64dc52e58af67cb20499)

Commit:
WIP. Update home/list screen.

Pulling in ideas from https://github.com/jfsiii/kibana/pull/2

Commit:
WIP. Update home/list screen.

Pulling in ideas from https://github.com/jfsiii/kibana/pull/2

Commit:
WIP. Home screen w/ plenty o'hacks but includes icons & better layout.

Commit:
WIP. Home screen w/ plenty o'hacks but includes icons & better layout.

Commit:
Use 4 column List view layout as in Figma. Left Title. Other 3 are cards

Commit:
Use 4 column List view layout as in Figma. Left Title. Other 3 are cards

Commit:
Disable lint rule while waiting on PR from kibana platform team.

Commit:
Disable lint rule while waiting on PR from kibana platform team.

Commit:
Use NP breadcrumbs. Add context w/ core & other deps. Add routes as dep.

Commit:
Use NP breadcrumbs. Add context w/ core & other deps. Add routes as dep.

Commit:
PLUGIN_ID -> PLUGIN.ID. Consistent w/most other plugins.

Commit:
PLUGIN_ID -> PLUGIN.ID. Consistent w/most other plugins.

Commit:
Move some repeated strings under PLUGIN namespace

Commit:
Move some repeated strings under PLUGIN namespace

Commit:
Less noisy (more idiomatic?) approach to contexts & hooks

Commit:
Less noisy (more idiomatic?) approach to contexts & hooks

Commit:
Add useBreadcrumbs hook. Views call useBreadcrumbs(arrayOfBreadcrumbs).

Commit:
Add useBreadcrumbs hook. Views call useBreadcrumbs(arrayOfBreadcrumbs).

Commit:
Detail page uses EuiPage & EuiPageBody like the List page does.

Commit:
Detail page uses EuiPage & EuiPageBody like the List page does.

joshdover committed:
Fix missing angular providers for map hacks (#42021)

jgowdyelastic committed:
[ML] Fixing file datavisualizer mapping overrides (#42047)

bartvanremortele committed:
[APM] Add support for very high durations (minutes and hours) (#41640)

dmlemeshko committed:
Functional tests: make sure clicked dashboard is opened (#42006)

* [page_objects/dashboard_page] make sure selected dashboard is opened

thomasneirynck committed:
[Maps] Format doc-ids as strings (#42011)

thomasneirynck committed:
fit only on visible features (#42020)

Commit:
Added test to verify revert license to basic. (#41716)

* Added test to verify revert license to basic.

* Removed smoke test tag. Doesn't need to run on FF.

mshustov committed:
Export public http interfaces (#42001)

* re-export http types

* update docs

Commit:
flip sql statement expansion logic (#41531)

Bargs committed:
Remove sort, move, and remove buttons from Context app's table header (#42026)

Prior to #41259 we intentionally did not show the sort, move, and remove column buttons in the doc table in the Context app. Context doesn't pass in handlers for these actions and in the old angular directive we conditionally showed the buttons depending on whether those handlers were passed in. So in this PR I've simply mimicked that behavior in the new React component.

thomasneirynck committed:
[Maps] fix EMSClient import (#42073)

simianhacker committed:
[Infra UI] Use Kibana date format for chart headers (#41715)

- Add tooltip props with headerFormatter to Metrics Explorer charts
- Add tooltip props with headerFormatter to Metric detail charts
- Closes #40011

simianhacker committed:
[Infra UI] Change contextual menu position on infrastructure view (#41810)

* [Infra UI] Change contextual menul position on infrastructure view

* Adding ConditionalToolTip wrapper component

clintandrewhall committed:
[Canvas] Update object-path-immutable; type AxisConfig and Con… (#41086)

* Initial WIP

* Migrate to latest object-path-immutable; type arguments

* Fix Storybook Snapshot test; add snapshots

* Revert storyshots test change

* Adding SeriesStyle conversion + stories

* Addressing feedback; updating against master

* Update snapshots

sorenlouv committed:
[APM] Remove confusing transaction group abstraction (#41886)

alexwizp committed:
[Table Visualization] (Step 2) Shim new platform - renaming table_vis -> vis_type_table (#41872)

* [Table Visualization] (Step 2) Shim new platform - renaming table_vis -> vis_type_table

* fix PR comment

Commit:
[Region Map] Shim new Platform (#40966)

* Migrate Region Map to New Platform

* Replace Vis Factory

* fix plugin name - work in progress

* code cleanup

* fix tests for tile_map

* fix PR comments

* fix PR comments

Commit:
Add Hero image. Vertically center copy and push image to bottom.

Line-height, color and some other items are not to spec yet.

Using inline styles while we're iterating. Will extract to CSS / Sass soon.

Commit:
Add Hero image. Vertically center copy and push image to bottom.

Line-height, color and some other items are not to spec yet.

Using inline styles while we're iterating. Will extract to CSS / Sass soon.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9663/)

sorenlouv committed:
[APM] Fix transaction distribution redirection bug (#42104)

Commit:
Filter manager improvements \ bug fixes (#41999)

* Filter manager improvements

 - Avoid firing update \ fetch events twice from filter manager, by making filter state manager use getFilters rathre than storing its own previous filters.
- Don't fire fetch event when only disabled filters were changed
- Replaced onlyDisabledChanged implementation with a more readable one (karma tests pass!)
- Exported onlyDisabledChanged from data
- Deleted unused only_state_changed

* Delete empty afterEach

sorenlouv committed:
[APM] Local filters preparation: move uiFilters encoding to server and update routes (#41233)

Commit:
Factor <HeroCopy> & <HeroImage> out of <Header>

Commit:
Remove header background. Add grey bottom border.

Commit:
Remove header background. Add grey bottom border.

Commit:
Fix i18n errors. Prior bad merge of 81153c2

Commit:
Fix i18n errors. Prior bad merge of 81153c2

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9694/)

Commit:
Use PLUGIN.TITLE now that it has translated string

Commit:
Use PLUGIN.TITLE now that it has translated string

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9697/)

jgowdyelastic committed:
[ML] Fix jobs list loading when auto refresh is off (#42076)

sulemanof committed:
[Vis: Default editor] EUIficate pie chart options tab (#41901)

* EUIficate pie chart options tab

* Fix tests

* Size titles down to xs

* Use FormattedMessage

maryia-lapata committed:
[Vis Editor] EUIfication of agg and agg-group directives (#40866)

* Create default_editor_agg.tsx

* Create default_editor_agg_group

* Apply drag and drop

* Remove unused dragula dependency

* Remove old mocha tests

* Add ts for state

* Update functional tests

* Update touched condition

* Apply styles for accordion button content

* Apply truncate for agg description

* Remove unused styles

* Separate common props

* Move aggGroupNamesMap to agg_group.js

* Update _sidebar.scss

* Pass schemas prop

* Prevent scroll bar and add space

* Remove unused min from stats

* Add OnAggParamsChange type

* Show error as an icon

* Update background color

* Update title size

* Remove Schema.deprecate since it's not used

Commit:
Update heading hierarchy for instructions page (#41970)

vitalics committed:
[TSVB] [Functional] unskip flaky test (#41766)

* unskip flaky test

mikecote committed:
Modify task manager getRetryDelay to be more custom (#42064)

* Modify task manager getRetryDelay to be more custom

* Ensure intervals don't die

* Update comment

* Update comment pt2

* Fix type check

* Intervals to not have retry logic

* Make test easier to read

* Modify docs

* Rename confusing variable

* Apply PR feedback

* Change error to task timeout error

* Re-add comment

* Fix type check

walterra committed:
[ML] Data Frames: Update stats data structure. (#42117)

Fixes UI regressions caused by breaking API changes.

gregkalapos committed:
[APM] Update .NET Agent instructions (#42124)

* Update src/legacy/core_plugins/kibana/server/tutorials/apm/instructions/apm_agent_instructions.js

.NET Agent:
- removed beta warning
- updated the text with the updated pacakge and `UseAllElasticApm()` method
- fixed 1 minor typo

* Update translations files

sorenlouv committed:
[APM] Move apm links into apm folder (#42118)

* [APM] Move APM links into apm folder and reuse TransactionLink component

* Prefer `filter` over `must` in queries

* Fix transaction type switcher

andrewvc committed:
[Uptime][Docs] Fix ref to index aliases in docs (#42094)

Commit:
Use snake case for image name

Commit:
Use snake case for image name

flash1293 committed:
No reload on changes to disabled filters in dashboard (#41144)

walterra committed:
[ML] Fix display of progress bar. (#42158)

Fixes the check whether to display the empty progress bar for the stopped state. (This bug would display both progress bar, the moving one and the empty one when a continuous job was started)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9811/)

kertal committed:
Add modal with details to shard error toast (#41649)

* Add details to shard error toast

* Implement failure modal with lots of details

* Add jest tests

* Update core API change because of the new modal Prop 'className'

walterra committed:
[ML] Migrates single metric viewer to React. (#41739)

Migrates the overall page of Single Metric Viewer to React.

Commit:
[ES-UI] Authorization lib to control access to feature & app sections (#40150)

alvarezmelissa87 committed:
[ML] Data Frames - Creation wizard editor for source config (#41937)

* Advanced editor for source config

* Separate editor for source and pivot

* Reset query when toggle back to KQL if changes have been made. Show warning

* update snapshot

* retain source config editor state. display in code block

* ensure apply button set correctly

* update source config editor helptext

* update request and step_define_summary test

* update step_define_summary snapshot

* No source editor switch if saved search.

Zacqary committed:
[Logs UI] Increase keypress delay in functional tests (#41777)

* [Logs UI] Increase keypress delay in functional tests

* Re-enable moveLogColumn test

* Use charByChar instead of arbitrary key delays

* Remove contradictory expect statement

* Revert to 500ms keypress delay

dgieselaar committed:
[APM] Make sure tooltips for metric charts are synced (#42139)

Closes #42138

dgieselaar committed:
[APM] Add react-hooks lint rules for APM folder, fix dependencies (#42129)

* [APM] Add react-hooks lint rules for APM folder, fix dependencies

Closes #42128.

* Validate useFetcher dependencies as well

* Add useFetcher hook; move eslint config to kibana eslint config

alisonelizabeth committed:
[WATCHER] remove validation for email action body attribute (#42009)

alexwizp committed:
[TSVB] Group by "Everything" fails to calculate data with "Overall" metrics (#42074)

* [TSVB] Group by "Everything" fails to calculate data with "Overall" metrics

Closes: #42015

* add tests

* fix grammar

* hasPipelineAggregations -> hasSiblingPipelineAggregation

Commit:
Use pipes for communication with chrome to avoid networking snafus (#42097)

thomasneirynck committed:
[Maps] Add test coverage for layer-syncing (#42089)

cjcenizal committed:
Extract useRequest and sendRequest from SR into es-ui-shared plugin (#40777)

- Refactor SR's consumption to send UI metric telemetry in the promise chain, to uncouple the request functions from telemetry.
- Refactor useRequest to return isInitialRequest instead of tracking polling state.
- Rename request callback to sendRequest.
- Rename loading flag to isLoading.
- Rename changeInterval callback to setRequestInterval.
- Rename interval parameter to pollIntervalMs.
- Remove initialData from return value.
- Preserve complete response shape instead of removing everything except the data property.
- Send a request when path, body, or method changes.
- Don't send a request when the interval changes.
- Remove setRequestInterval from return value.
- Expect the consumer to change the interval config in order to set it to a new value.
- Refactor internals so that calling sendRequest resets the interval.
- Migrate Watcher to use shared request library.
- Rename onSuccess to deserializer and use it to process data.
- Rename updateInterval function to scheduleRequest.
- Don’t watch method parameter.

joshdover committed:
[new-platform] Add `legacy` property to NavLinks registered by… (#41301)

poffdeluxe committed:
Mocking Date.now for storyshots tests and re-enabling previously flaky tests (#42184)

jasonrhodes committed:
Reverts submit on blur for infra logs query bar (#41968)

mw-ding committed:
[Code] remove the disableIndexScheduler flag (#42179)

mw-ding committed:
[Code] Initial server side refactor to shim the new platform server API (#41192)

* [Code] introduce a RequestFacade interface to decouple hapi.Request

* Add a RequestQueryFacade

* completely remove hapi dependencies from code

* Refactor init.ts to comply with the new platform plugin API

* more refactoring

* fix tests

* fix type errors

szabosteve committed:
[DOCS] Updates ML/anomaly detection terms in the Kibana guide (#41965)

Commit:
Move non-migratable Angular dependencies into directives folder  (#41638)

* Moved render_directive to directives folder

* Moved debounce to directives

* Moved bind to directives

* Moved watch multi into directives

* Moved listen to directives

zfy0701 committed:
[Code] Add localization for Code

jfsiii commented on x-pack/legacy/plugins/integrations_manager/public/data.ts:
```diff
@@ -5,26 +5,59 @@
  */
 
 import { npStart } from 'ui/new_platform';
+// disabling pending https://github.com/elastic/kibana/pull/42001
```
I see https://github.com/elastic/kibana/pull/42001 has landed, so I'll remove these comments

mikecote committed:
Expose spaces plugin function signatures in TypeScript (#42133)

* Expose spaces plugin function signatures

* Apply PR feedback

kobelb committed:
No longer registering timelion feature if it's disabled (#42193)

In 7.0, Timelion added a `timelion.ui.enabled` setting which hides
Timelion from being displayed as an app, and it's only visible within
Visualize. We did not adjust feature controls based on this new setting,
so it was always showing up as a Feature in both Roles and Spaces Management

Commit:
Merge branch 'master' of github.com:elastic/kibana into feature-integrations-manager

Commit:
Merge branch 'update-list-view' of github.com:jfsiii/kibana into update-list-view

Commit:
Remove comments disabling import warnings.

https://github.com/elastic/kibana/pull/42001 has landed

jfsiii commented:
Abandoning this due to issues after rebasing against `feature-integrations-manager` to get a desired. Will open a new PR.

jfsiii commented:
Replaced with https://github.com/elastic/kibana/pull/42260

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10001/)
